### PR TITLE
ENH: Automate deploy to PyPI

### PR DESCRIPTION
**Pull request by tbirdso**
Automatically deploys wheels to the Python Package Index on tag.

Depends on `PYPI_PASSWORD` secret.

Closes https://github.com/InsightSoftwareConsortium/itk-dreg/issues/9

### CODE REVIEW

**Comment by tbirdso** [comment]
@thewtex Would you please create a PyPI access token and add it as a `itk-dreg` repository secret? https://itkpythonpackage.readthedocs.io/en/master/Build_ITK_Module_Python_packages.html?highlight=twine#automate-pypi-package-uploads

**Review by thewtex — CHANGES_REQUESTED** [review]
@tbirdso thanks!

A few notes inline to transition to enable Trusted Publishing instead of using token / secrets, which is modern / preferred. We could not enable this in the [ITK Remote Module Action yet because Trusted Publishing is not supported in reusable workflows](https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/issues/68).

**Comment by tbirdso** [comment]
Hi @thewtex , I do not see your suggested changes on the PR?

File: .github/workflows/build-test-publish.yml [hunk 1]
```diff
@@ -57,3 +57,22 @@ jobs:
...
+      uses: actions/download-artifact@v3
+      with:
+        name: wheel
+        path: dist
+
+    - name: "Publish Python wheel to the Python Package Index"
+      uses: pypa/gh-action-pypi-publish@v1.8.10
+      with:
+        skip_existing: true
+        password: ${{ secrets.pypi_password }}
```
Comment (thewtex) [inline .github/workflows/build-test-publish.yml:78]: @tbirdso I added [Trusted Publishing](https://github.com/pypa/gh-action-pypi-publish) support on PyPI, so this should be removed.

but we should also add:

```
    environment:
      name: pypi
      url: https://pypi.org/p/<your-pypi-project-name>
    permissions:
      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
```
```suggestion
```

**Comment by thewtex** [comment]
@tbirdso sorry, not sure what happened -- added as a Single Comment.

File: .github/workflows/build-test-publish.yml (hunk 1 above)
Comment (tbirdso) [reply .github/workflows/build-test-publish.yml]: Great! Good to know. I've pushed a commit with the requested changes.

**Comment by tbirdso** [comment]
@thewtex Pushed a last commit on my way out the door, feel free to push to my dev branch if there are other changes required to get this through. Thanks!

**Review by thewtex — APPROVED** [review]
:rocket:

**Pull request merged**